carbonserver/carbonserver/api/routers/organizations.py

```diff
@@ -53,9 +50,9 @@ def read_organization(
     "/organizations/", tags=ORGANIZATIONS_ROUTER_TAGS, status_code=status.HTTP_200_OK
 )
 @inject
-def read_organizations(
+def list_organizations(
     organization_service: OrganizationService = Depends(
         Provide[ServerContainer.organization_service]
     ),
 ) -> List[Organization]:
-    return organization_service.list_organization()
+    return organization_service.list_organizations()
```

Comment:
Get /organizations/ give me an error:
```
carbonserver_1  | INFO:     172.22.0.1:48934 - "GET /organizations/?token=jessica HTTP/1.1" 500 Internal Server Error
carbonserver_1  | ERROR:    Exception in ASGI application
carbonserver_1  | Traceback (most recent call last):
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 371, in run_asgi
carbonserver_1  |     result = await app(self.scope, self.receive, self.send)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/uvicorn/middleware/proxy_headers.py", line 59, in __call__
carbonserver_1  |     return await self.app(scope, receive, send)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/fastapi/applications.py", line 199, in __call__
carbonserver_1  |     await super().__call__(scope, receive, send)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/applications.py", line 112, in __call__
carbonserver_1  |     await self.middleware_stack(scope, receive, send)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/middleware/errors.py", line 181, in __call__
carbonserver_1  |     raise exc from None
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/middleware/errors.py", line 159, in __call__
carbonserver_1  |     await self.app(scope, receive, _send)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/exceptions.py", line 82, in __call__
carbonserver_1  |     raise exc from None
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/exceptions.py", line 71, in __call__
carbonserver_1  |     await self.app(scope, receive, sender)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 580, in __call__
carbonserver_1  |     await route.handle(scope, receive, send)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 241, in handle
carbonserver_1  |     await self.app(scope, receive, send)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 52, in app
carbonserver_1  |     response = await func(request)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/fastapi/routing.py", line 214, in app
carbonserver_1  |     raw_response = await run_endpoint_function(
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/fastapi/routing.py", line 151, in run_endpoint_function
carbonserver_1  |     return await run_in_threadpool(dependant.call, **values)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/starlette/concurrency.py", line 40, in run_in_threadpool
carbonserver_1  |     return await loop.run_in_executor(None, func, *args)
carbonserver_1  |   File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
carbonserver_1  |     result = self.fn(*self.args, **self.kwargs)
carbonserver_1  |   File "/usr/local/lib/python3.8/dist-packages/dependency_injector/wiring.py", line 597, in _patched
carbonserver_1  |     result = fn(*args, **to_inject)
carbonserver_1  |   File "./carbonserver/api/routers/organizations.py", line 58, in list_organizations
carbonserver_1  |     return organization_service.list_organizations()
carbonserver_1  |   File "./carbonserver/api/services/organization_service.py", line 28, in list_organizations
carbonserver_1  |     organizations: List[Organization] = self._repository.list_organizations()
carbonserver_1  |   File "./carbonserver/api/infra/repositories/repository_organizations.py", line 62, in list_organizations
carbonserver_1  |     orgs.append(self.map_sql_to_schema(org))
carbonserver_1  |   File "./carbonserver/api/infra/repositories/repository_organizations.py", line 76, in map_sql_to_schema
carbonserver_1  |     return Organization(
carbonserver_1  |   File "pydantic/main.py", line 406, in pydantic.main.BaseModel.__init__
carbonserver_1  | pydantic.error_wrappers.ValidationError: 1 validation error for Organization
carbonserver_1  | api_key
carbonserver_1  |   none is not an allowed value (type=type_error.none.not_allowed)

```

Reply:
It seems that an org is saved without an api key (and pydantic does not seem to like it). Did you flush your db before reloading the API ?

Reply:
Bug was : when bulk inserting with Alembic you must reference the same columns for all inserts.
Admin API key was not set thus Community org api key was not taken into account.
Users could not join community organisation as api key was not set. This should be better now, thanks for the help.